client/hostinfo_unix.cpp

```diff
@@ -1584,6 +1584,45 @@ int HOST_INFO::get_memory_info() {
     return 0;
 }
 
+#if LINUX_LIKE_SYSTEM
+// gather libc version from the system installed ldd binary
+// a usual first line looks like: ldd (Debian GLIBC 2.24-11+deb9u1) 2.24
+// where the last part is copied to version and the string between parenthesis
+// is copied to extra_info
+//
+// return 0 if at least version could be found (extra_info may remain empty)
+// return 1 if ldd couldn't be opened or no version infromation was found
+//
+int get_libc_version(string& version, string& extra_info) {
+    char buf[1024];
+    string strbuf;
+    FILE* f = popen("PATH=/usr/bin:/bin:/usr/local/bin ldd --version 2>&1", "r");
+    if (f) {
+        fgets(buf, sizeof(buf), f);
+        pclose(f);
```

Comment:
You need to check return value here. If `ldd` doesn't exist `pclose` returns an error code and `buf` will contain something like `sh: 1: ldd: not found`.

Reply:
Thanks, fixed in 11b5569